sc2ts/inference.py

```diff
@@ -1971,45 +1982,77 @@ def asdict(self):
         return dataclasses.asdict(self)
 
 
-def rematch_recombinant(base_ts, recomb_ts, node_id, num_mismatches):
-    # dataset, recomb_ts, match_db, strain, *, num_mismatches, deletions_as_missing=False):
-    # print("Rematch for", strain)
-    if recomb_ts.nodes_flags[node_id] & core.NODE_IS_RECOMBINANT == 0:
-        raise ValueError(f"node {node_id} is not a recombinant")
-
-    if node_id < base_ts.num_nodes:
-        raise ValueError(f"node {node_id} already present")
-
-    assert base_ts.num_sites == recomb_ts.num_sites
-
-    haplotype = np.zeros(recomb_ts.num_sites, dtype=np.int8)
-    for var in recomb_ts.variants(
+def create_sample(ts, node_id):
+    haplotype = np.zeros(ts.num_sites, dtype=np.int8)
+    for var in ts.variants(
         samples=[node_id], alleles=tuple(core.IUPAC_ALLELES), isolated_as_missing=False
     ):
         haplotype[var.site.id] = var.genotypes[0]
-    sample = Sample("tmp", haplotype=haplotype)
 
-    group_id = recomb_ts.node(node_id).metadata["sc2ts"]["group_id"]
+    sample = Sample("tmp", haplotype=haplotype)
+    group_id = ts.node(node_id).metadata["sc2ts"]["group_id"]
     # Find a causal sample to get the original match. This is just to get a rough bound
     # on the path likelihood so it doesn't really matter which one.
-    tree = recomb_ts.first()
+    tree = ts.first()
     for u in tree.nodes(node_id, "timedesc"):
         if u != node_id:
-            md = recomb_ts.node(u).metadata["sc2ts"]
+            md = ts.node(u).metadata["sc2ts"]
             if md.get("group_id", None) == group_id and tree.is_sample(u):
                 hmm_match = md["hmm_match"]
                 break
 
     sample.hmm_match = HmmMatch.fromdict(hmm_match)
 
+    return sample
+
+def move_mutations_above_recombinant(ts, re_node):
```

Comment:
Might as well use the [existing function](https://github.com/jeromekelleher/sc2ts/blob/104defae81eef8ac4bbc6a5b2c097e7b8ca4bf77/sc2ts/inference.py#L2408) which does it for all RE nodes, but is reasonably fast.